src/pages/ImportData.js

```diff
@@ -119,17 +121,40 @@ const ImportData = () => {
       }],
       ['snapshot', async () => {
         snapshots && snapshots.length > 0 ?
-          await Promise.allSettled(_.map(({ title, id }) => Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, title), snapshots)) :
-          await Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, snapshotName)
-        notify('success', 'Snapshot imported successfully.', { timeout: 3000 })
+          await Promise.allSettled(
+            _.map(({ title, id, description }) => {
+              // Replace all whitespace characters with _
+              // Then replace all non alphanumeric characters with nothing
+              const normalizedTitle = title.replace(/\s/g, '_').replace(/[^A-Za-z0-9\s-_]/g, '')
+              return Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, normalizedTitle, description)
+            }, snapshots)
+          ).then(async responses => {
+            success = !_.some(({ status }) => status === 'rejected', responses)
+
+            if (!success) {
+              const normalizedResponses = await Promise.all(_.map(async ({ status, reason }) => {
+                const reasonJson = reason ? await reason.json() : '{}'
+                const message = JSON.parse(reasonJson.message || '{}').message
+                return { status, message }
+              }, responses))
+              setSnapshotResponses(normalizedResponses)
+              notify('error', `There was a problem importing ${snapshots.length > 1 ? 'some of these snapshots' : 'this snapshot'}`, { timeout: 3000 })
+            }
+          }) :
+          await Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, snapshotName).then(() => {
```

Comment:
This await appears to be a no-op.

Reply:
I think it should continue to await to confirm whether or not this succeeds or fails before redirecting the user to the workspace